compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java

```diff
@@ -345,7 +345,8 @@ public enum DiagnosticErrorCode implements DiagnosticCode {
     ERROR_INVALID_BINDING_PATTERN("BCE0677", "error.invalid.binding.pattern"),
     ERROR_RESOURCE_PATH_CANNOT_BEGIN_WITH_SLASH("BCE0678", "error.resource.path.cannot.begin.with.slash"),
     REST_PARAMETER_CANNOT_BE_INCLUDED_RECORD_PARAMETER("BCE0679",
-            "error.rest.parameter.cannot.be.included.record.parameter")
+            "error.rest.parameter.cannot.be.included.record.parameter"),
+    ERROR_INVALID_ANNOTATIONS_REST_MEMBER("BCE0680", "error.annotations.not.allowed.for.rest.member"),
```

Comment:
We usually use the same enum member name in the dotted format as the `messageKey`.  Also, "not allowed for rest member" is generic. Shall we say,  "tuple rest descriptor"?
```suggestion
    ERROR_ANNOTATIONS_NOT_ALLOWED_FOR_TUPLE_REST_DESCRIPTOR("BCE0680", "error.annotations.not.allowed.for.tuple.rest.descriptor"),
```

When adding an entry here, you also need to add an entry at `syntax_diagnostic_message.properties` for the `messageKey`.